daemon/emer-daemon.c

```diff
@@ -548,6 +566,89 @@ add_events_to_builders (GVariant       **events,
     }
 }
 
+static gboolean
+parse_event_id (const gchar *unparsed_event_id,
+                uuid_t       parsed_event_id)
+{
+  gint parse_failed = uuid_parse (unparsed_event_id, parsed_event_id);
+  if (parse_failed != 0)
+    {
+      g_warning ("Attempt to parse UUID \"%s\" failed. Make sure you created "
+                 "this UUID with uuidgen -r. You may need to sudo apt-get "
+                 "install uuid-runtime first.", unparsed_event_id);
+      return FALSE;
+    }
+
+  return TRUE;
+}
+
+static gboolean
+report_invalid_data_in_cache_on_idle (CacheMetricEventData *callback_data)
+{
+  EmerDaemon *self = callback_data->daemon;
+  const char *event_id = callback_data->event_id;
+  GVariant *payload = callback_data->payload;
+  GVariant *actual_payload = NULL;
+
+  gint64 relative_time;
+  if (!emtr_util_get_current_time (CLOCK_BOOTTIME, &relative_time))
+    {
+      g_critical ("Getting relative timestamp failed.");
+      goto free;
+    }
+
+  uuid_t parsed_event_id;
+  if (!parse_event_id (event_id, parsed_event_id))
+    {
+      g_critical ("Could not parse event ID");
+      goto free;
+    }
+
+  GVariantBuilder uuid_builder;
+  get_uuid_builder (parsed_event_id, &uuid_builder);
+  GVariant *event_id_variant = g_variant_builder_end (&uuid_builder);
+
+  if (payload != NULL)
+    {
+      actual_payload = payload;
+    }
+  else
+    {
+      GVariant *unboxed_variant = g_variant_new_boolean (FALSE);
+      GVariant *empty_payload = g_variant_new_variant (unboxed_variant);
+      actual_payload = g_variant_ref_sink (empty_payload);
```

Comment:
I'm not completely sure either, I borrowed this [idiom from emtr_event_recorder_init](https://github.com/endlessm/eos-metrics/blob/27e4194ed5ca1c5966261ef1fed5fa182a4285a8/eosmetrics/emtr-event-recorder.c#L172), as that's how it constructs the empty payload for some reason, and assumed that was the right thing to do as I can't see the server side of things. But I'm happy to write it simpler if possible